Update requests requirement

TopdRob opened the pull request:
Update request requirement from version v2.14.2 to v2.18.4

## Description:
I've checked the dependencies with https://requires.io/github/TopdRob/home-assistant/requirements/?branch=dev to check if I could update the dependencies. I started with requests package to update it from v2.14.2 to v2.18.4

### Changelog (copied from https://github.com/requests/requests/blob/master/HISTORY.rst)
2.18.4 (2017-08-15)
+++++++++++++++++++

**Improvements**

- Error messages for invalid headers now include the header name for easier debugging

**Dependencies**

- We now support idna v2.6.

2.18.3 (2017-08-02)
+++++++++++++++++++

**Improvements**

- Running ``$ python -m requests.help`` now includes the installed version of idna.

**Bugfixes**

- Fixed issue where Requests would raise ``ConnectionError`` instead of
  ``SSLError`` when encountering SSL problems when using urllib3 v1.22.

2.18.2 (2017-07-25)
+++++++++++++++++++

**Bugfixes**

- ``requests.help`` no longer fails on Python 2.6 due to the absence of
  ``ssl.OPENSSL_VERSION_NUMBER``.

**Dependencies**

- We now support urllib3 v1.22.

2.18.1 (2017-06-14)
+++++++++++++++++++

**Bugfixes**

- Fix an error in the packaging whereby the ``*.whl`` contained incorrect data
  that regressed the fix in v2.17.3.

2.18.0 (2017-06-14)
+++++++++++++++++++

**Improvements**

- ``Response`` is now a context manager, so can be used directly in a ``with`` statement
  without first having to be wrapped by ``contextlib.closing()``.

**Bugfixes**

- Resolve installation failure if multiprocessing is not available
- Resolve tests crash if multiprocessing is not able to determine the number of CPU cores
- Resolve error swallowing in utils set_environ generator


2.17.3 (2017-05-29)
+++++++++++++++++++

**Improvements**

- Improved ``packages`` namespace identity support, for monkeypatching libraries.


2.17.2 (2017-05-29)
+++++++++++++++++++

**Improvements**

- Improved ``packages`` namespace identity support, for monkeypatching libraries.


2.17.1 (2017-05-29)
+++++++++++++++++++

**Improvements**

- Improved ``packages`` namespace identity support, for monkeypatching libraries.


2.17.0 (2017-05-29)
+++++++++++++++++++

**Improvements**

- Removal of the 301 redirect cache. This improves thread-safety.


2.16.5 (2017-05-28)
+++++++++++++++++++

- Improvements to ``$ python -m requests.help``.

2.16.4 (2017-05-27)
+++++++++++++++++++

- Introduction of the ``$ python -m requests.help`` command, for debugging with maintainers!

2.16.3 (2017-05-27)
+++++++++++++++++++

- Further restored the ``requests.packages`` namespace for compatibility reasons.

2.16.2 (2017-05-27)
+++++++++++++++++++

- Further restored the ``requests.packages`` namespace for compatibility reasons.

No code modification (noted below) should be neccessary any longer.

2.16.1 (2017-05-27)
+++++++++++++++++++

- Restored the ``requests.packages`` namespace for compatibility reasons.
- Bugfix for ``urllib3`` version parsing.

**Note**: code that was written to import against the ``requests.packages``
namespace previously will have to import code that rests at this module-level
now.

For example::

    from requests.packages.urllib3.poolmanager import PoolManager

Will need to be re-written to be::

    from requests.packages import urllib3
    urllib3.poolmanager.PoolManager

Or, even better::

    from urllib3.poolmanager import PoolManager

2.16.0 (2017-05-26)
+++++++++++++++++++

- Unvendor ALL the things!

2.15.1 (2017-05-26)
+++++++++++++++++++

- Everyone makes mistakes.

2.15.0 (2017-05-26)
+++++++++++++++++++

**Improvements**

- Introduction of the ``Response.next`` property, for getting the next
  ``PreparedResponse`` from a redirect chain (when ``allow_redirects=False``).
- Internal refactoring of ``__version__`` module.

**Bugfixes**

- Restored once-optional parameter for ``requests.utils.get_environ_proxies()``.

2.14.2 (2017-05-10)
+++++++++++++++++++

**Bugfixes**

- Changed a less-than to an equal-to and an or in the dependency markers to
  widen compatibility with older setuptools releases.

### Vizio
Because the requests package no longer includes third party libraries they we had to update the Vizio implementation as well. Therefore this is also added to this pull request.

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

TopdRob committed:
Update request requirement

Update request requirement from version v2.14.2 to v2.18.4

homeassistant commented:
Hi @TopdRob, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#9876).

Once you do that we will be able to review and accept this pull request.

Thanks!

Danielhiversen commented:
Have you tested this change locally?

TopdRob commented:
Yes, according to these instructions: https://home-assistant.io/developers/development_environment/

I have my test environment running on Raspbian

homeassistant commented:
Hi @mgillespie, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#9876).

Once you do that we will be able to review and accept this pull request.

Thanks!

homeassistant commented:
Hi @mgillespie, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#9876).

Once you do that we will be able to review and accept this pull request.

Thanks!

TopdRob committed:
Fix dependency vizio integration

3rd patry packages were removed from requests. Changed dependency from requests to urllib3

TopdRob committed:
forgot =

forgot = when adding the requirement

homeassistant commented:
Hi @mgillespie, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#9876).

Once you do that we will be able to review and accept this pull request.

Thanks!

homeassistant commented:
Hi @mgillespie, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#9876).

Once you do that we will be able to review and accept this pull request.

Thanks!

TopdRob committed:
re-run script/gen_requirements_all.py

re-run script/gen_requirements_all.py

balloob commented on homeassistant/components/media_player/vizio.py:
```diff
@@ -32,7 +32,7 @@
 from homeassistant.helpers import config_validation as cv
 
-REQUIREMENTS = ['pyvizio==0.0.2']
+REQUIREMENTS = ['pyvizio==0.0.2', 'urllib3==1.22']
```
You should not have to add a dependency on urllib3, it should get installed as a dependency of requests 2.16 and later (when they unvendored all the things).

TopdRob committed:
Unvendoring urllib3 from requests

In v2.16.0 and newer of requests they unverdored urllib3.

TopdRob committed:
Merge branch 'dev' of https://github.com/TopdRob/home-assistant into dev

TopdRob committed:
undefined name 'InsecureRequestWarning'

TopdRob committed:
Removed requirement to 'urllib3==1.22

TopdRob committed:
removed import requests

TopdRob commented:
I'm lost :-(

Danielhiversen commented on homeassistant/components/media_player/vizio.py:
```diff
@@ -75,11 +75,11 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
-        import requests
-        from requests.packages.urllib3.exceptions import InsecureRequestWarning
+        from requests.packages import urllib3
+        urllib3.exceptions.InsecureRequestWarning
```
Try to remove this line

TopdRob replied:
Thanks!

TopdRob committed:
removed urllib3.exceptions.InsecureRequestWarning

removed urllib3.exceptions.InsecureRequestWarning travis lint
